minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/ShutdownHook.java

```diff
@@ -55,10 +55,12 @@ public void run() {
 
         System.out.println("Initiating shutdown of bootstrap change ingestors...");
         ConfigurationChangeCoordinator notifier = runner.getChangeCoordinator();
-        try {
-            notifier.close();
-        } catch (IOException ioe) {
-            System.out.println("Could not successfully stop notifier due to " + ioe);
+        if (notifier != null) {
```

Comment:
Hmm, didn't see that in mine in testing.  Although I can certainly see how this would happen.  

Could initialize the listener/ingestors before starting the actual flow.  This might even make more sense logically.